homeassistant/config_entries.py

```diff
@@ -939,7 +947,19 @@ async def async_remove(self, entry_id: str) -> dict[str, Any]:
                 )
             )
 
-        return {"require_restart": not unload_success}
+        result: dict[str, Any] = {"require_restart": not unload_success}
+        for (domain, call) in self._post_remove_calls:
+            if domain == entry.domain and (data := await call(self.hass, entry)):
+                result.update(data)
+        return result
+
+    def async_add_post_remove_call(
```

Comment:
Not such a fan of this name as it doesn't explain it's meant to add extra info to the result.

I wonder if we can do this simpler by adding a WS API to check if a config entry uses application credentials, and call that from the frontend before processing the remove call. That way we don't need any of this callback registrations.

Reply:
Sure, happy to explore that. So the idea is that frontend would do a specific check for an application credentials specific WS API before deleting? Sure, sounds good.  I don't yet quite have the intuition about when frontend should depend on specific/arbitrary core integrations and where it shouldn't, so this is fine to me.

Reply:
I've removed and replaced with a websocket. Let me know if this is what you had in mind, then i'll take another pass on the frontend PR.

Reply:
I'm going to pro-actively start working on the FE side of this, assuming it's going in the right direction in order, to make sure we have progress on this by next release, given the severity of the usability issues for this.  (soft ping)